jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -196,6 +203,18 @@ default HttpFields asImmutable()
         return HttpFields.build(this).asImmutable();
     }
 
+    /**
+     * <p>Returns an immutable version of this {@code HttpFields} instance, possibly optimized for random field
+     * access via {@link Map} implementations. If this {@code HttpFields} is mutable, then subsequent modifications
+     * are not reflected in the instance returned by this method.</p>
+     *
+     * @return an {@link HttpFields} instance, which may be optimized for random access.
+     */
+    default HttpFields asMappedAccess()
```

Comment:
Do we really need this?
It's a lot of code, a lot of tests, and I fail to see the use case.

We have already determined that `Set` is _in practice_ slower than `List` in almost all cases, despite the theory.

Reply:
Probably not.    This was more an exercise in improving the testing of HttpFields and to see what random access could look like.   Now that it is written, I think we could offer it, with no harm as it is entirely optional.
When I have a moment, I'll do some extras JMH tests with EnumSet.   Or maybe we just offer the class as a wrapper?

Reply:
@lorban @sbordet 
Probably not.    This was more an exercise in improving the testing of HttpFields and to see what random access could look like.   Now that it is written, I think we could offer it, with no harm as it is entirely optional.
When I have a moment, I'll do some extras JMH tests with EnumSet....  oh did it...

```
Benchmark                                             Mode  Cnt          Score          Error  Units
HashMapVsEnumMapBenchmark.testEnumMapBuildAndLookup  thrpt    3  238432890.283 ± 79828131.575  ops/s
HashMapVsEnumMapBenchmark.testHashMapBuildAndLookup  thrpt    3   58008556.754 ±  4374032.136  ops/s
```

So EnumMap is an order of magnitude faster than HashMap.   I wonder if we should just always build a `EnumMap` of known headers and let the others be done with the list.   The problem there is that misses will always need to iterate.

Hmmmm

Reply:
I'm tempted to believe that iterating the list will have comparable perf to looking up in a hash map, assuming the list stays small.

After a second thought, I don't mind this extra `asMappedAccess()` but if it doesn't bring any measurable improvement, it should probably go.

Reply:
> I'm tempted to believe that iterating the list will have comparable perf to looking up in a hash map, assuming the list stays small.

We measured multiple times that a list is faster than a map/set up until about 250 elements, which is never going to be the case for the number of HTTP headers.

I don't see the point of adding more APIs to maintain, and I'm positive any difference will be buried into the noise of all the rest.
I'd keep it simpler.

Reply:
I expanded the JMH to test enum field look up vs a list, but also using the full HttpHeader enum rather than a nominal little enum.  List is still way better, even for small sets of fields and missed lookup.  So I'm closing this.